services/wallet/history/service_test.go

```diff
@@ -271,11 +272,68 @@ func Test_entriesToDataPoints(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name: "multipleAddresses",
+			args: args{
+				chainIDs: []uint64{1, 2},
+				data: []*entry{
+					{
+						chainID:   2,
+						balance:   big.NewInt(5),
+						timestamp: 1,
+						address:   common.Address{1},
+					},
+					{
+						chainID:   1,
+						balance:   big.NewInt(6),
+						timestamp: 1,
+						address:   common.Address{2},
+					},
+					{
+						chainID:   1,
+						balance:   big.NewInt(1),
+						timestamp: 2,
+						address:   common.Address{1},
+					},
+					{
+						chainID:   1,
+						balance:   big.NewInt(2),
+						timestamp: 3,
+						address:   common.Address{2},
+					},
+					{
+						chainID:   1,
+						balance:   big.NewInt(4),
+						timestamp: 4,
+						address:   common.Address{2},
+					},
+				},
+			},
+			want: []*DataPoint{
+				{
+					Balance:   (*hexutil.Big)(big.NewInt(11)),
+					Timestamp: 1,
+				},
+				{
+					Balance:   (*hexutil.Big)(big.NewInt(7)),
```

Comment:
Why isn't `6` expected instead of the current `7`? As I see it, chain `2` doesn't change balance, so there is still balance `5` in there, and with the new value of `1`, for chain `1` I expect to have value `6` in here.

Reply:
You're right; the balance change is per chain and address identity.